notification-spring/src/main/java/org/duracloud/notification/SpringNotificationFactory.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ *     http://duracloud.org/license/
+ */
+package org.duracloud.notification;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+import javax.mail.MessagingException;
+
+import org.apache.commons.validator.routines.EmailValidator;
+
+import org.duracloud.common.error.DuraCloudRuntimeException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.mail.javamail.JavaMailSenderImpl;
+
+/**
+ * @author Shibo Liu
+ * Date: 07/03/19
+ */
+public class SpringNotificationFactory implements NotificationFactory {
+
+    private static final Logger log = LoggerFactory.getLogger(
+        SpringNotificationFactory.class);
+
+    private JavaMailSenderImpl emailService;
+    private Map<String, Emailer> emailerMap = new HashMap<String, Emailer>();
+    private String host;
+    private Integer port;
+
+    public SpringNotificationFactory(String host, Integer port) {
+        this.host = host;
+        this.port = port;
+    }
+
+    @Override
+    public void initialize(String username, String password) {
+        String logUserMsg = "{}";
```

Comment:
It's not clear why this defaults to `"{}"`. If username is null or empty, you want this to be an empty string, right? Why not just set it to `""`? As it is, in the case where username is null (see next comment), the log messages may look odd.